Feature [MIQ-2199]: Add support for date based partitions for hiveQL from Superset

shmsr opened the pull request:
### CATEGORY
Currently, it supports up to partitions like Date/ Hour. Please do review it.

The feature was implemented keeping in mind the best practices while making partitions in hive. Now superset will support queries to hive which are partitioned on the basis of date (%Y-%m-%d)

Ref: [Best Practices for Hive Partitions by Date](https://community.cloudera.com/t5/Support-Questions/Best-Pratices-for-Hive-Partitioning-especially-by-Date/m-p/112525)

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

shmsr committed:
Add support for date based partitions for hiveQL

arpit-agarwal commented on superset/hive_query.py:
```diff
@@ -95,49 +96,114 @@ def get_partitioned_whereclause(_st, _en, gran_seconds, time_partitions):
+
+    if grain == 0:
+        # Partition = Date
+        part_time_seq = list()
```
This seems redundant in this if block

subham-sarkar-guavus replied:
Okay, I'll shift it.

subham-sarkar-guavus replied:
Done that. Also committed some changes which handle some corner cases (important).

shmsr committed:
Handle case when no filter is applied [st, en] for date based partition

shmsr committed:
Handle cases when reserved keywords are used as column names

> Avoids error like "Error while compiling statement: FAILED: ParseException line 5:9 Failed to recognize predicate \'column_name\'. Failed rule: \'identifier\' in table or column identifier"
> Problem arised when query used reserved keywords without tilde quoting

shmsr committed:
Refactoring

shmsr committed:
Fix UTs for hive query partition related changes

shmsr committed:
Remove redundant code block
